moto/route53/responses.py

```diff
@@ -30,32 +30,47 @@ class Route53(BaseResponse):
     def list_or_create_hostzone_response(self, request, full_url, headers):
         self.setup_class(request, full_url, headers)
 
+        # Set these here outside the scope of the try/except
+        # so they're defined later when we call create_hosted_zone()
+        vpcid = None
+        vpcregion = None
         if request.method == "POST":
             elements = xmltodict.parse(self.body)
-            if "HostedZoneConfig" in elements["CreateHostedZoneRequest"]:
-                comment = elements["CreateHostedZoneRequest"]["HostedZoneConfig"][
-                    "Comment"
-                ]
+            zone_request = elements["CreateHostedZoneRequest"]
+            if "HostedZoneConfig" in zone_request:
+                zone_config = zone_request["HostedZoneConfig"]
+                comment = zone_config["Comment"]
                 try:
                     # in boto3, this field is set directly in the xml
-                    private_zone = elements["CreateHostedZoneRequest"][
-                        "HostedZoneConfig"
-                    ]["PrivateZone"]
+                    private_zone = zone_config["PrivateZone"]
                 except KeyError:
-                    # if a VPC subsection is only included in xmls params when private_zone=True,
-                    # see boto: boto/route53/connection.py
-                    private_zone = "VPC" in elements["CreateHostedZoneRequest"]
+                    # if a VPC subsection is only included in xmls
+                    # params when private_zone=True, see boto:
+                    # boto/route53/connection.py
+                    private_zone = "VPC" in zone_request
             else:
                 comment = None
                 private_zone = False
 
-            name = elements["CreateHostedZoneRequest"]["Name"]
+            if private_zone == "true":
```

Comment:
```suggestion
            if private_zone:
```
Which means this can be simplified

Reply:
Hi Bert,

I actually had this as `if private_zone:` originally. However, it doesn't actually return a boolean. `private_zone` was set to `"false"`, which was still evaluating as `True`, which is why I ended up changing it to the expanded comparison using strings.  I'll double check it and post the output from the debugger.

Thanks!
Paul

Reply:
@bblommers, here's the trace from the debugger:

```> /Users/pll/dev/github/moto/moto/route53/responses.py(47)list_or_create_hostzone_response()
-> private_zone = zone_config["PrivateZone"]
(Pdb) zone_config["PrivateZone"]
'true'
(Pdb) type(zone_config["PrivateZone"])
<class 'str'>
```

So something somewhere is setting this to a string, not a boolean, unfortunately...

Reply:
@bblommers 

Any thoughts on the above ? Good enough to leave as a string comparison? Or should do something else?

Reply:
Fair enough - a string comparison makes sense then :+1: